### PR TITLE
Feat: Oracle migration script

### PR DESCRIPTION
**Pull request by haythem96**
# Feat: Oracle migration script

## High Level Description

This PR add a script to migrate expiry prices from an old oracle module to new Oracle module.
This PR also add `migrateOracle()` in the Oracle module that allow anyone to push prices for an assets.
The function implementation is a bit hacky as it:
- Does not verify if those expiries timestamp are old
- Can set malicious prices if owner is compromised

The above notes are not critical issues IMO, to mitigate any risk:
- deploy Oracle and run the migration script
- Set `migrated` to true by calling `endMigration()`
- Add new Oracle address to AddressBook as the last step

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [ ] Is your code up to date with the spec? 
- [ ] Have you added your tests to the testing doc?

### CODE REVIEW

**Comment by antoncoding** [comment]
why not set an array of expiry and read the value from the old oracle?

**Comment by haythem96** [comment]
> why not set an array of expiry and read the value from the old oracle?

I thought about that, I thought it may consume alot of gas... what do u think ?

**Comment by antoncoding** [comment]
> I thought about that, I thought it may consume alot of gas... what do u think ?

I think it's much lower risk of messing up compared to writing off-chain script to do that. If we only migrate expiry that has otoken created, it's probably just 40~ reads, so it might be okay? 

I don't have really strong preference tbh.

File: scripts/migrateOracle.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    let expiriesToMigrate = []
+    let pricesToMigrate = []
+
+    for(let i = 0; i < otokenLength.toFixed(); i++) {
+      let otokenAddress = await factory.otokens(i)
+      let otokenDetails = await (await Otoken.at(otokenAddress)).getOtokenDetails()
+      let expiry = new BigNumber(otokenDetails[4])
+
+      if((otokenDetails[0] == options.asset || otokenDetails[1] == options.asset || otokenDetails[2] == options.asset) && (expiry.toNumber() < currentTimeStamp)) {
+        let priceToMigrate = new BigNumber((await oldOracle.getExpiryPrice(options.asset, expiry.toString()))[0])
```
Comment (antoncoding) [inline scripts/migrateOracle.js:40]: Why not use `const` for these variables

Comment (antoncoding) [reply scripts/migrateOracle.js:40]: I think we need a throw an error when the price is 0.

File: scripts/migrateOracle.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    
+    let expiriesToMigrate = []
+    let pricesToMigrate = []
+
+    for(let i = 0; i < otokenLength.toFixed(); i++) {
+      let otokenAddress = await factory.otokens(i)
+      let otokenDetails = await (await Otoken.at(otokenAddress)).getOtokenDetails()
+      let expiry = new BigNumber(otokenDetails[4])
+
+      if((otokenDetails[0] == options.asset || otokenDetails[1] == options.asset || otokenDetails[2] == options.asset) && (expiry.toNumber() < currentTimeStamp)) {
```
Comment (antoncoding) [inline scripts/migrateOracle.js:39]: we can also include expiry that's already be set right? Maybe use a Set to keep track of all the expiry that's already pushed into the `expiriesToMigrate` array

Comment (haythem96) [reply scripts/migrateOracle.js:39]: Added that

File: scripts/migrateOracle.js (hunk 1 above)
Comment (haythem96) [reply scripts/migrateOracle.js:40]: let's skip this for now, as we know when we run the script that all expired options already have a pushed price.

File: scripts/migrateOracle.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    const currentTimeStamp = Math.floor(Date.now() / 1000);
+    const otokenLength = new BigNumber(await factory.getOtokensLength())
+    
+    let expiriesToMigrate = []
+    let pricesToMigrate = []
+
+    for(let i = 0; i < otokenLength.toFixed(); i++) {
+      let otokenAddress = await factory.otokens(i)
+      let otokenDetails = await (await Otoken.at(otokenAddress)).getOtokenDetails()
+      let expiry = new BigNumber(otokenDetails[4])
```
Comment (antoncoding) [inline scripts/migrateOracle.js:37]: Can you use `const` instead of `let`?

**Review by antoncoding — APPROVED** [review]

**Review by aparnakr — APPROVED** [review]

**Pull request merged**